python/ray/util/sgd/torch/examples/gat_dgl.py

```diff
@@ -0,0 +1,274 @@
+                        h = layer(block, (h, h_dst)).flatten(1)
+                    else:
+                        h = layer(block, (h, h_dst)).mean(1)
+                        h = h.log_softmax(dim=-1)
+                    y[output_nodes] = h.cpu()
+                x = y
+            pred = y
+        labels = g.ndata['labels']
+        train_acc, val_acc, test_acc = compute_acc(pred[train_nid], labels[train_nid]), \
+                                       compute_acc(pred[val_nid], labels[val_nid]), \
+                                       compute_acc(pred[test_nid], labels[test_nid])
+        metrics = {
+            "num_samples": pred.size(0),
+            "val_acc": val_acc.item(),
+            "test_acc": test_acc.item()
+        }
+        meter_collection.update(metrics, n=metrics.pop("num_samples", 1))
+        status = meter_collection.summary()
+        return status
+
+
+def run(num_workers=1, use_gpu=False, num_epochs=2):
+    trainer = TorchTrainer(
+        training_operator_cls=CustomTrainingOperator,
+        num_workers=num_workers,
+        config={
+            "lr": 1e-3,
+            "batch_size": 1024,
+        },
+        use_gpu=use_gpu,
+        backend="nccl",
+        # use_tqdm=True,
+    )
+
+    for i in range(num_epochs):
+        trainer.train()
+    testResults = trainer.validate()
+    trainer.shutdown()
+    print(testResults)
+    print("success!")
+
+
+# Use ray.init(address="auto") if running on a Ray cluster.
+if __name__ == '__main__':
+    argparser = argparse.ArgumentParser("multi-gpu training")
+    argparser.add_argument('--num-epochs', type=int, default=2)
+    argparser.add_argument('--n-hidden', type=int, default=128)
+    argparser.add_argument('--n-layers', type=int, default=2)
+    argparser.add_argument('--n-heads', type=int, default=4)
+    argparser.add_argument('--fan-out', type=str, default='10,25')
+    argparser.add_argument('--batch-size', type=int, default=1024)
+    argparser.add_argument('--lr', type=float, default=0.001)
+    argparser.add_argument('--feat-drop', type=float, default=0.)
+    argparser.add_argument('--attn-drop', type=float, default=0.)
+    argparser.add_argument('--negative-slope', type=float, default=0.2)
+    argparser.add_argument(
+        '--num-workers',
+        type=int,
+        default=0,
+        help="Number of sampling processes. Use 0 for no extra process.")
+    argparser.add_argument(
+        '--dashboard-host',
+        type=str,
+        default='127.0.0.1',
+        help="The host to bind the dashboard server to.")
+    argparser.add_argument(
+        '--dashboard-port',
+        type=int,
+        default=8265,
+        help="The port to bind the dashboard server to.")
+    args = argparser.parse_args()
+    ray.init(
+        dashboard_host=args.dashboard_host, dashboard_port=args.dashboard_port)
+    ###connect to started ray cluster
+    # ray.init(address='auto', _redis_password='5241590000000000')
```

Comment:
You can take in the address as an arg that defaults to None. And then you can just have a single ray.init like `ray.init(address=args.address, ...)`